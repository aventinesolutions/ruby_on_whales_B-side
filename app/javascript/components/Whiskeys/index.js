import React from 'react';
import { Query } from 'react-apollo';
import Whiskey from '../Whiskey';
import gql from 'graphql-tag';
import './styles.scss';

const WhiskeysQuery = gql`
  {
    whiskeys {
      id
      description
      price
      photoUrl
    }
  }`;

export default () => (
	<Query query={WhiskeysQuery}>
		{({ data, loading }) => (
			<div className='whiskeys-container'>
				{loading ? 'loading ...' : data.whiskeys.map(whiskey => <Whiskey key={whiskey.id} {...whiskey} />)}
			</div>
		)}
	</Query>
);
