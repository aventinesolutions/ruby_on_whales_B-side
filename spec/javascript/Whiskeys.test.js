import React from 'react';
import gql from 'graphql-tag';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Whiskeys from '../../app/javascript/components/Whiskeys';

const query = gql`
  {
    search(filter: {accountId: "c6fc5f72-6d50-4111-856e-c07eadbaef6b",
      text: "handcrafted",
      ratingsFilter: ""
    }) {
      id
      title
      description
      price
      photoUrl
      ratings {
        id
        accountId
        quality
        stars
      }
    }
  }
`;

let output;

beforeEach(() => {
  output = shallow(
    <Whiskeys query={query}/>
  );
});

describe('Whiskeys', () => {
  it('renders a Whiskeys component', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
