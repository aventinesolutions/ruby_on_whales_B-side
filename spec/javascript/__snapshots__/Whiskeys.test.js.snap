// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Whiskeys renders a Whiskeys component 1`] = `
<Query
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": undefined,
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "filter",
                    },
                    "value": Object {
                      "fields": Array [
                        Object {
                          "kind": "ObjectField",
                          "name": Object {
                            "kind": "Name",
                            "value": "accountId",
                          },
                          "value": Object {
                            "block": false,
                            "kind": "StringValue",
                            "value": "c6fc5f72-6d50-4111-856e-c07eadbaef6b",
                          },
                        },
                        Object {
                          "kind": "ObjectField",
                          "name": Object {
                            "kind": "Name",
                            "value": "text",
                          },
                          "value": Object {
                            "block": false,
                            "kind": "StringValue",
                            "value": "handcrafted",
                          },
                        },
                        Object {
                          "kind": "ObjectField",
                          "name": Object {
                            "kind": "Name",
                            "value": "ratingsFilter",
                          },
                          "value": Object {
                            "block": false,
                            "kind": "StringValue",
                            "value": "",
                          },
                        },
                      ],
                      "kind": "ObjectValue",
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "search",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "title",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "description",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "price",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "photoUrl",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "ratings",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "accountId",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "quality",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "stars",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 295,
        "start": 0,
      },
    }
  }
>
  <Component />
</Query>
`;
